packages/grafana-traces/src/pyroscope/datasource.ts

```diff
@@ -0,0 +1,28 @@
+import { Observable } from 'rxjs';
+
+import { AbstractQuery, CoreApp, DataQueryRequest, DataQueryResponse, ScopedVars } from '@grafana/data';
+import { DataSourceWithBackend } from '@grafana/runtime';
+
+import { PyroscopeDataSourceOptions, Query, ProfileTypeMessage } from './types';
+
+export abstract class PyroscopeDataSource extends DataSourceWithBackend<Query, PyroscopeDataSourceOptions> {
```

Comment:
Bit confused by this. Couldn't we just change the dependency to be an interface that then pyroscope will have to conform to? That seams like more reasonable dependency structuring to be than creating an abstract class here.